### PR TITLE
'Settled' and 'cancelled' events should display in IOUDetails Modal

### PR DESCRIPTION
**Pull request by Julesssss**
### Details

As of the latest [IOU Details issue](https://github.com/Expensify/Expensify/issues/158811), only IOU transaction events are displayed in the Modal, but not 'Settled' and  'Cancelled' events. This PR refactors the previous logic, enabling all reportComments of type IOU to be shown in the IOUDetails Modal.

Screenshot 2021-05-20 at 15 37 53

This also fixes [a bug](https://github.com/Expensify/Expensify/issues/164704) caused by incorrect Number to String Javascript conversion.

### Fixed Issues

Fixes https://github.com/Expensify/Expensify/issues/164703
Fixes https://github.com/Expensify/Expensify/issues/164704

### Tests

_NOTE: If you see the `This action was taken on old code and is not supported anymore` message, the IOU was created from out-of-date Auth code. Please ensure Auth is pulled and rebuilt, then create a new IOU to perform the test._

1. As **userA**, create an IOU via 'Request Money' to **userB**
2. Within the chatReport with **userB**, you should see the following message: 
Screenshot 2021-05-20 at 16 10 45
3. _Note that the full Preview Component might not display, [due to this bug](https://github.com/Expensify/Expensify/issues/164530)_
4. Repeat step two, creating a second IOU transaction against the same user
Screenshot 2021-05-20 at 16 18 49
5. Now, we need to cancel one of the transactions (QA should skip this step)
5b. To find the `**reportID**`, click View Details and note the second route param
Screenshot 2021-05-20 at 16 24 28
5c. To locate the `**transactionID**`, open the Chrome dev tools Application/Local Storage tab, and search the `reportID` from 5b. Locate `reportIOUs_XYZ` where XYZ is the reportID.
5d. Use one of the transactionID values in the API request in 5e
Screenshot 2021-05-20 at 16 25 48
5e.  Make the following API call, replacing authToken, email, accountId, **reportID**, & **transactionID**
```
curl --request POST \
  --url 'https://expensify-jules.ngrok.io/api?command=RejectTransaction' \
  --header 'Content-Type: multipart/form-data; boundary=---011000010111000001101001' \
  --cookie 'email=user8%2540expensify.com; accountID=162; authToken=BCFB...04FAB' \
  --form reportID=1182 \
  --form transactionID=2212712
```
5f. You should see a new 'Declined/Cancelled the $12.34 request' message, depending on which used rejected the transaction
Screenshot 2021-05-20 at 16 41 02
6. Now, log out and log back in as **userB** 
7. Once logged in, refresh the page to [avoid this bug](https://github.com/Expensify/Expensify/issues/164467)
8. Navigate to the chatReport with **userA**
Screenshot 2021-05-20 at 16 37 45

9. Click 'View Details'', or 'Pay'
10. Settle the IOU by tapping the  'I'll settle up elsewhere' button
Screenshot 2021-05-20 at 16 54 36
11. You should see an event for each IOU action that we performed:
    - Requested $100.00
    - Requested $1.00
    - Declined the $1.00 request 
    - Settled up elsewhere

### QA Steps

Run above tests, but **SKIP STEP 5!**, as this is not yet possible.

### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

#### Web
Screenshot 2021-05-25 at 16 06 06

#### Mobile Web
Screenshot 2021-05-25 at 16 27 14

#### Desktop
Screenshot 2021-05-25 at 16 28 47

#### iOS
Simulator Screen Shot - iPhone 11 - 2021-05-25 at 16 15 37

#### Android
Screenshot_1621955303

### CODE REVIEW

**Comment by Julesssss** [comment]
CC @iwiznia you may be interested in reviewing, but no pressure

**Review by MonilBhavsar — DISMISSED** [review]

File: src/pages/iou/IOUTransactions.js [hunk 1]
```diff
@@ -18,65 +17,36 @@ const propTypes = {
...
+    reportActions,
+    chatReportID,
+    iouReportID,
+}) => (
+    <View style={[styles.mt3]}>
+        {_.map(reportActions, (reportAction) => {
+            if (reportAction.actionName === 'IOU'
+                && reportAction.originalMessage.IOUReportID === iouReportID) {
+                // Cancelled/declined actions reuse the original transactionID, so we must make the key unique
+                const transactionKey = `${reportAction.originalMessage.IOUTransactionID}-
```
Comment (iwiznia) [inline src/pages/iou/IOUTransactions.js:36]: This is not ensured to be unique, can we use the reportAction's sequence number instead?

**Review by iwiznia — DISMISSED** [review]

File: src/pages/iou/IOUTransactions.js (hunk 1 above)
Comment (Julesssss) [reply src/pages/iou/IOUTransactions.js]: Yep, much better. Done 👍

**Comment by Julesssss** [comment]
Ready for another review @MonilBhavsar @iwiznia:
- Merged main
- Switched the list item key to sequenceNumber

**Review by MonilBhavsar — APPROVED** [review]
Screenshots left to be added.

**Comment by Julesssss** [comment]
Added the screenshots

**Review by iwiznia — APPROVED** [review]

**Pull request merged**